docs/guides/blocks/sync-tables/two-way.md

```diff
@@ -0,0 +1,300 @@
+      });
+      // Wait for all of the jobs to finish .
+      let completed = await Promise.allSettled(jobs);
+
+      // For each update, return either the updated row or an error if the
+      // update failed.
+      let results = completed.map(job => {
+        if (job.status === "fulfilled") {
+          return job.value;
+        } else {
+          return job.reason;
+        }
+      });
+      return {
+        results: results,
+      };
+    }
+  },
+});
+```
+
+When processing updates in batches be sure to return the same number of results as there were updates, in the same order. If an individual update fails, catch the exception and return it in the results array to indicate that the row update failed. Unhandled exceptions will prevent the entire batch from completing and won't give the user a clear indication of which row was the cause. See the [Handling errors](#handling-errors) section below for more information.
+
+
+## Property autocomplete {: #autocomplete}
+
+Similar to [parameter autocomplete][parameters_autocomplete], you can provide a suggested set of values for mutable properties, which will be shown to the user when they are editing a corresponding cell.
+
+<img src="../../../../images/two_way_autocomplete.png" srcset="../../../../images/two_way_autocomplete_2x.png 2x" class="screenshot" alt="Autocomplete in a cell.">
+
+The valid options are defined in the `autocomplete` field of the property, which can be either an array of static values or a function that generates them dynamically. An autocomplete function can access the value of other properties in the row via `context.propertyValues` and the current search string typed by the user via `context.search`.
+
+```ts
+const ShirtSchema = coda.makeObjectSchema({
+  properties: {
+    size: {
+      type: coda.ValueType.String,
+      autocomplete: ["S", "M", "L", "XL"],
+    },
+    color: {
+      type: coda.ValueType.String,
+      autocomplete: async function (context) {
+        let size = context.propertyValues.size;
+        let availableColors = await getColorsForSize(context, size);
+        return availableColors;
+      },
+    },
+    pattern: {
+      type: coda.ValueType.String,
+      autocomplete: async function (context) {
+        let search = context.search;
+        let patterns = await searchPatterns(context, search);
+        return patterns;
+      },
+    },
+  },
+  // ...
+});
+```
+
+For sync tables with dynamic schemas, you aren't allowed to define the search function directly on the property itself. Instead use the special value `AutocompleteType.Dynamic`, which tells Coda to call the sync table's `autocomplete` function. This is a single function that handles the auto-complete for all dynamic properties. It's defined directly on a dynamic sync table, or within the `dynamicOptions` of a regular sync table. The function can determine which property to provide autocomplete for by inspecting `context.propertyName`.
```

Comment:
Nit, optional, "aren't allowed" makes it sound a bit arbitrary, could frame it as "you aren't able to include code in dynamic responses" or something.

Reply:
Changed to "aren't able", but I don't want to get to much into "why" since it's complicated and really doesn't matter.